Use `tags` and `maintainers` instead of `site` and `agency`

lightandluck opened the pull request:
Places where changes were made:

1. The important changes and dictionary exports. I also updated them to reflect the format of current analysts sheets with Scanner links
2. The page listing
3. How the api queries for pages

lightandluck commented:
@Mr0grog I accidentally added a change from production to `important changes` and `dictionary` while testing. Can you revert? Sorry! I thought I was pointed at localhost. Staging-db isn't deployed with newest changes yet, so I was hopping back and forth between localhost and production to make sure my changes worked. The environment banner shows if we're pointed at staging, but doesn't make a distinction for localhost or production. I'll add an issue for it.

`page_uuid`: bbb97c8e-609b-4a9d-b50e-15b3220d4d19
`change`: 133aefaa-2c96-4925-8aeb-769f9290d94a..db33508a-5ce3-4592-80b8-a8dd5a78faf6

Mr0grog commented:
Will look into fixing. Added to the sheets or DB or both?

Also please never ever ever ever point your local app at production. If it’s causing a problem. Please yell at me to get off my butt and fix edgi-govdata-archiving/web-monitoring#98 first (next time).

Mr0grog commented on src/services/web-monitoring-db.js:
```diff
@@ -55,12 +55,27 @@ const storageLocation = 'WebMonitoringDb.token';
 
 /**
- * @typedef ApiResponse
+ * @typedef {Object} ApiResponse
```
FWIW, the `{type}` part of a `@typedef` [is supposed to be optional](http://usejsdoc.org/tags-typedef.html). It’s totally fine to add it, though — no need to change this back.

(The main reason to use it is to suggest that a type *isn’t* just an object, but a particular kind of `string` or `number` or other specific type. e.g. `@typedef {number} Milliseconds` so you can `@param {Milliseconds} delay` when defining a function.)

lightandluck replied:
I added it to be consistent with how all the other objects were defined. I think it makes a lot more sense to add a type for the reason you described. `{Object}` is so generic, it doesn't provide any useful information. 

Seems like we agree here, so I removed `{Object}` from all the other type definitions.

lightandluck commented:
😓 Absolutely noted. I will not do that again!

It was only added to DB. I was pointed at my own test sheets.

Mr0grog commented:
(DB is fixed)

lightandluck commented on server/sheet-data.js:
```diff
@@ -83,24 +83,24 @@ function addChangeToDictionary (data) {
     '',
     // UUID
-    `${data.from_version.uuid}..${data.to_version.uuid}`,
+    data.to_version.uuid,
```
Ah, that makes sense to make the distinction between the analysts' sheets and the annotated changes sheet. Changed it back.

lightandluck commented on server/sheet-data.js:
```diff
@@ -319,6 +319,18 @@ function formatDate (date) {
+function formatSites (tags) {
+  const isSite = /^site:/;
+  return tags
+    .filter(tagging => isSite.test(tagging.name))
```
Sweet! 👍

lightandluck commented on src/__mocks__/simple-page.json:
```diff
@@ -65,5 +63,36 @@
+      "uuid": "5954f070-c1b2-4a0d-9e97-42974b8875a0",
+      "name": "site:NOAA - ncei1.noaa.gov",
+      "assigned_at": "2018-02-19T23:10:17.853Z"
+    },
```
Just me being lazy creating mock data, but you're right, it should not be like that. I changed it around randomly.

lightandluck commented on src/__mocks__/simple-page.json:
```diff
@@ -65,5 +63,36 @@
+    }
   ]
-}
+}
```
Ah, sorry, didn't understand the importance of this until doing some reading. Installed and cool that you're maintaining one of the plugins 😄 . I'm unsure if I added back in correctly. Please double-check.

lightandluck commented on src/components/__tests__/page-list.test.jsx:
```diff
@@ -0,0 +1,37 @@
+    const pageList = shallow(
+      <PageList
+        pages={null}
+        onSearch={null}
```
True, removed

lightandluck commented on src/components/page-list.jsx:
```diff
@@ -126,3 +125,11 @@ function debounce (func, delay) {
+  const isSite = /^site:/;
+  return tags
+    .filter(tagging => isSite.test(tagging.name))
+    .map(tagging => tagging.name.replace(isSite, ''))
```
Yea, I was a bit confused because the server and app use 2 different module styles so didn't think it possible, but after some reading, realized I could go with CommonJS to get it to work. (There is some Babel magic happening there that I don't quite understand.)

I changed `formatters.js` to use `exports.foo` instead of `export foo`. This gave me linting errors about `exports not being defined`, so I added `/*eslint-env node*/`. Unsure if that's the right way to treat that error? Could use some help here 😕

Mr0grog replied:
👍 I think you  probably want `/* eslint-env commonjs */`, but `node` works fine, too.

Mr0grog replied:
(But otherwise, yeah, you’re doing the right thing! The alternative is to edit `src/.eslintrc.json`, but I think we want to keep this to just the files where it’s needed, so I would have done the same as you.)

lightandluck commented on src/services/web-monitoring-api.js:
```diff
@@ -151,7 +151,7 @@ export default class WebMonitoringApi {
   _getPagesByDomains (domains, dateRange, extraQuery) {
     const fetches = domains.map(domain => {
-      const query = Object.assign({site: domain}, extraQuery);
+      const query = Object.assign({'tags[]': `site:${domain}`}, extraQuery);
```
That is really awesome and works perfectly!

lightandluck commented on src/services/web-monitoring-db.js:
```diff
@@ -14,17 +14,17 @@ const storageLocation = 'WebMonitoringDb.token';
  * @property {Version} [latest]
  * @property {Version[]} [versions]
+ * @property {Maintainers[]} maintainerships
+ * @property {Tags[]} taggings
```
Yup, I mixed it up. I understand the reason for the distinction and naming, but it's definitely not intuitive at first glance.

lightandluck commented:
@Mr0grog Let me know how things look now. Thanks!

Mr0grog commented:
Looks fantastic to me; only two *suuuuuuper minor* notes that you can change and merge or leave as-is:

- In `formatters.js`, the env should probably be `commonjs` (AKA browser code w/ `require()`, `module`, and `exports`). You could also do `node,browser`, I think.
- For the mock tags and maintainers, the names should all be different, too (duplicated names are not allowed, even if with different upper/lower-case).

lightandluck commented:
Cool, switched the `env`. 
Good to know! The tag names were actually different, one was `ncei.` and the other `ncei1.`, but I switched in other names so it would be easier to tell.

In it goes!

Mr0grog commented:
🎉
